servicex/__init__.py

```diff
@@ -43,6 +45,29 @@
 from servicex.transformer_manager import TransformerManager
 
 
+class ServiceXFormatter(logging.Formatter):
+    """
+    Need a custom formatter to allow for logging with request ids that vary.
+    Normally log messages are "level instance component request_id msg" and
+    request_id gets set by initialize_logging but we need a handler that'll let
+    us pass in the request id and have that embedded in the log message
+    """
+
+    def format(self, record: logging.LogRecord) -> str:
+        """
+        Format record with request id if present, otherwise assume None
+
+        :param record: LogRecord
+        :return: formatted log message
+        """
+
+        if hasattr(record, "requestId"):
+            return super().format(record)
```

Comment:
Move this duplicated line outside of `if` statement

Reply:
Updated